pkgs/applications/virtualization/singularity/template.nix

```diff
@@ -0,0 +1,163 @@
+{ lib
+, util-linux
+, gpgme
+, openssl
+, libuuid
+, bash
+, coreutils
+, which
+, makeWrapper
+, cryptsetup
+, squashfsTools
+, nvidia-docker # For nvidia-container-cli
+, glibc # For ldconfig needed by nvidia-container-cli
+, go
+, buildGoModule
+, pname
+, version
+, src
+, projectName # "apptainer" or "singularity"
+  # These cannot be correctly override with `overrideAttrs`
+, vendorSha256 ? null
+, deleteVendor ? false
+, proxyVendor ? false
+, doCheck ? true
+, withoutSuid ? true # Required to run containers without suid
```

Comment:
Is the intention that singularity use userns "by default" and suid on nixos?

Reply:
@SomeoneSerge AFAIK, the `programs.singularity` NixOS module is meant to enable user to have a setuid-ed singularity package, which is only achievable with admin intervention.

As a unprivileged build, the plain Nix package of singularity should contain the build option to opt out the suid feature, or it won't be able to run packages without `sudo`. This was a bug fixed in this PR.

Reply:
🤔 should we then expose `apptainerWithoutSuid` attributes so that these are cached by `hydra`?

Reply:
> thinking should we then expose `apptainerWithoutSuid` attributes so that these are cached by `hydra`?

I don't get it. Do you mean exposing a version of apptainer built with `withoutSuid = false`?